### PR TITLE
Ldapi , trust boundary, XSS rule for OWASP benchmark

### PR DESCRIPTION
**Pull request by colleend**
improves the xss no-direct-response-writer rule (changes it to taint mode) and creates a trust boundary rule and ldap injection rule for the OWASP java benchmark -- this improves the score from: 
* -6%-> 7% for xss
* 0% -> 11% for trust boundary
* 0% -> 9% for ldapi

_To test:_
Run `semgrep --test .` on the `java/lang/security` folder and check that all tests pass.

Run 
`semgrep -f ~/Desktop/semgrep_repos/semgrep-rules/java/lang/security/audit --json > results/Benchmark_1.2-semgrep.json`
`createScorecards.sh`

and check the scorecards for the benchmark.

### CODE REVIEW

File: java/lang/security/audit/tainted-ldapi-from-http-request.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+rules:
+  - id: tainted-ldapi-from-http-request
```
Comment (kurt-r2c) [inline java/lang/security/audit/tainted-ldapi-from-http-request.yaml:2]: we should try a version of this on pre-2.15 log4j 🙃

Comment (colleend) [reply java/lang/security/audit/tainted-ldapi-from-http-request.yaml:2]: ohh that would be interesting

File: java/lang/security/audit/tainted-session-from-http-request.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+rules:
+  - id: tainted-session-from-http-request 
+    message: >-
+      Detected input from a HTTPServletRequest going into a session command.
```
Comment (minusworld) [inline java/lang/security/audit/tainted-session-from-http-request.yaml]: I'm not certain I follow what this rule detects. What kind of object comes back from request.getSession()? Is it like a mapping of user sessions? What can I do with that session?

**Review by minusworld — CHANGES_REQUESTED** [review]
Left one comment on the message for getSession

File: java/lang/security/audit/tainted-session-from-http-request.yaml (hunk 2 above)
Comment (colleend) [reply java/lang/security/audit/tainted-session-from-http-request.yaml]: It's for the "trust boundary violation" portion of the OWASP java benchmark -- there's functions for getSession such as `setAttribute` that sets a parameter for that session (and that's where the issue lies). I'll add that to the message for clarity.

**Review by minusworld — APPROVED** [review]
lgtm!

**Pull request merged**